src/sparseml/modifiers/pruning/wanda/pytorch.py

```diff
@@ -0,0 +1,179 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import Any, Dict, Iterable, List, Optional, Tuple
+
+import torch
+
+from sparseml.core.model.base import ModifiableModel
+from sparseml.core.state import State
+from sparseml.modifiers.obcq.utils.helpers import cache_attention_inputs
+from sparseml.modifiers.pruning.wanda.base import WandaPruningModifier
+from sparseml.modifiers.pruning.wanda.utils.layer_compressor import WandaLayerCompressor
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class WandaPruningModifierPyTorch(WandaPruningModifier):
+    """
+    PyTorch implementation of WandaPruningModifier
+    """
+
+    model: Optional[ModifiableModel] = None
+    device_: str = "cuda:0"
+    layer_prefix_: Optional[str] = None
+    prunen_: Optional[int] = None
+    prunem_: Optional[int] = None
+
+    def on_initialize(self, state: State, **kwargs) -> bool:
+        """
+        Initialize and run the WANDA algorithm on the current state
+
+        :param state: session state storing input model and calibration data
+        """
+        self._validate_layerwise_sparsity()
+
+        self.initialize_wanda(state, **kwargs)
+
+        # run wanda on calibration data
+        self.apply_wanda(dataloader=state.data.calib)
+        torch.cuda.empty_cache()
```

Comment:
Curious, what's the use of this method?

Reply:
https://pytorch.org/docs/stable/generated/torch.cuda.empty_cache.html

TLDR; it's used to free up some GPU memory